src/unlimited.cpp

```diff
@@ -346,13 +346,16 @@ void settingsToUserAgentString()
 {
     BUComments.clear();
 
-    double ebInMegaBytes = (double)excessiveBlockSize / 1000000;
     std::stringstream ebss;
-    ebss << std::fixed << std::setprecision(1) << ebInMegaBytes;
+    ebss << (excessiveBlockSize / 100000);
     std::string eb = ebss.str();
-    std::string eb_formatted;
-    eb_formatted = (eb.at(eb.size() - 1) == '0' ? eb.substr(0, eb.size() - 2) : eb); // strip zero decimal
-    BUComments.push_back("EB" + eb_formatted);
+    eb.insert(eb.size() - 1, ".", 1);
+    if (eb.substr(0, 1) == ".")
```

Comment:
The intent is avoid any kind of risk due to floating point arithmetic since it is consensus related (`BUComments` are used also to signal in the coinbase string). I'd prefer to avoid float/double in this cases. But I could live with your proposed solution either. let see what @gandrewstone will say.